### PR TITLE
feat : add watermark pdf script

### LINKED ISSUE #236: PDF watermarker
## Script Title - Should able to watermark pdfs

**what will change** - Adding script

## Instructions

- Create a new folder for your script and file/folder name should be appropriate.
- Create a README.md in your folder for program Instructions
- add requirements.txt if needed

Please add/delete options that are not relevant.

- [x] Adding New Code

### Programming Language

- [x] Python

Happy Coding,

### PR DESCRIPTION
**Pull request by Hardik0307**
# Pull Request Template

**script name** - Should able to watermark pdfs

**What have you Changed**

added folder add_watermark_pdf with script and files.

Fixes #236 

### Self Check(Tick After Making pull Request)

- [x] This issue was assigned to me.
- [x] One Change in one Pull Request
- [x] My file is in proper folder (Name of folder should be in lowercase with no space in between)
- [x] I am following clean code and Documentation and my code is well linted with flake8.
- [x] I have added README.md and requirements.txt with my script

### CODE REVIEW

**Comment by pawangeek** [comment]
Hey, is there any possibility to customize watermarks? (Just simple text like the name of person)

**Comment by Hardik0307** [comment]
> Hey, is there any possibility to customize watermarks? (Just simple text like the name of person)

yes, @pawangeek.
we can have customized pdf I have mentioned in the README file too.

**Comment by pawangeek** [comment]
Okay, so we are adding a pdf with a watermark on every page of our main pdf? Correct

**Comment by Hardik0307** [comment]
> Okay, so we are adding a pdf with a watermark on every page of our main pdf? Correct

Yes.

**Comment by pawangeek** [comment]
Just want that end-to-end user able to create that watermark pdf with own name initially, followed by merging

**Comment by Hardik0307** [comment]
> Just want that end-to-end user able to create that watermark pdf with own name initially, followed by merging
> Is it possible to do so?

yes working on that.
Thanks for feedback.

**Comment by Hardik0307** [comment]
@pawangeek,
There is *one* issue that customized watermark deal with every individual image file. (depend on size)
for multipage pdf, it will be not productive.

and nowadays own watermark can be generated from *vscode* or *adobe* or [office](https://support.microsoft.com/en-us/office/use-a-custom-watermark-954cd296-7dab-4731-b5dd-84f9fa15bcd3).

File: add_watermark_pdf/script.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+inputFile = open('Demo.pdf', 'rb')
+pdfReader = PyPDF2.PdfFileReader(inputFile)
+inputFirstPage = pdfReader.getPage(0)
+
+pdfWatermarkReader = PyPDF2.PdfFileReader(open('watermark.pdf', 'rb'))
+inputFirstPage.mergePage(pdfWatermarkReader.getPage(0))
+pdfWriter = PyPDF2.PdfFileWriter()
+pdfWriter.addPage(inputFirstPage)
+
+for pageNum in range(1, pdfReader.numPages):
```
Comment (pawangeek) [inline add_watermark_pdf/script.py:18]: It's already dealing with every page of multipage PDF

File: add_watermark_pdf/script.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@add_watermark to pdf
+@py code
+"""
+import PyPDF2
+
+inputFile = open('Demo.pdf', 'rb')
+pdfReader = PyPDF2.PdfFileReader(inputFile)
+inputFirstPage = pdfReader.getPage(0)
+
+pdfWatermarkReader = PyPDF2.PdfFileReader(open('watermark.pdf', 'rb'))
```
Comment (pawangeek) [inline add_watermark_pdf/script.py:13]: And just want that user is able to make own watermark pdf by just entering the name (That simply replace TOP SECRET do not accidentally ... with own name)

**Review by pawangeek — CHANGES_REQUESTED** [review]

File: add_watermark_pdf/script.py (hunk 1 above)
Comment (Hardik0307) [reply add_watermark_pdf/script.py:18]: Agreed but when it comes to image it become tedious.

**Comment by pawangeek** [comment]
I am saying just to edit text. No image allowed

> Hey, is there any possibility to customize watermarks? (**Just simple text like the name of person**)
See

File: add_watermark_pdf/script.py (hunk 2 above)
Comment (Hardik0307) [reply add_watermark_pdf/script.py:13]: I know the purpose but the generation of the image with watermark text takes place together.
I find the solution but it was not *feasible*.

Hence I didn't implement it.

Comment (pawangeek) [reply add_watermark_pdf/script.py:13]: Try to implement that, we make it feasible later

Comment (Hardik0307) [reply add_watermark_pdf/script.py:13]: Okay. :+1:

**Comment by pawangeek** [comment]
@Hardik0307 do you come up with a solution?

**Comment by Hardik0307** [comment]
> @Hardik0307 do you come up with a solution?

Not really.

**Pull request closed without merging**